fs/socket/socket.c

```diff
@@ -192,16 +196,20 @@ FAR struct socket *file_socket(FAR struct file *filep)
   return NULL;
 }
 
-FAR struct socket *sockfd_socket(int sockfd)
+int sockfd_socket(int sockfd, FAR struct socket **socketp)
 {
   FAR struct file *filep;
 
+  *socketp = NULL;
```

Comment:
I'm not sure if this is needed. If we decide to keep it, then let's place it before `return -EBADF;`